src/v/ssx/abort_source.h

```diff
@@ -38,6 +38,8 @@ class sharded_abort_source {
         });
     }
 
```

Comment:
You aren't using sharded_about_source any more, right? So this could go away?

Reply:
Yeah, will be removing this commit when I squash everything.